packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -205,17 +225,87 @@ export const SampleFinderSection: FC<Props> = memo(props => {
             });
 
             onFilterClose();
-            updateFilters(filterChangeCounter + 1, newFilterCards);
+            updateFilters(filterChangeCounter + 1, newFilterCards, !currentView?.entityId, true);
 
             api.query.incrementClientSideMetricCount(SAMPLE_FILTER_METRIC_AREA, 'filterModalApply');
         },
-        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType]
+        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType, cardDirty, currentView]
+    );
+
+    const loadSearch = useCallback(
+        async (view: FinderReport) => {
+            let cardJson = null;
+
+            if (view.isSession) cardJson = sessionStorage.getItem(getLocalStorageKey());
+            else if (view.reportId) cardJson = await loadFinderSearch(view);
+            if (!cardJson) return;
+
+            const finderSessionData = searchFiltersFromJson(cardJson);
+            const newFilters = finderSessionData.filters;
+            if (!newFilters) return;
+
+            updateFilters(filterChangeCounter + 1, newFilters, false, view.isSession);
+            setShowSaveViewDialog(false);
+            setCurrentView(view);
+        },
+        [filterChangeCounter]
+    );
+
+    const onSaveComplete = useCallback((view: FinderReport) => {
+            setShowSaveViewDialog(false);
+            setViewDirty(false);
+            setCurrentView(view);
+            setSavedViewChangeCounter(savedViewChangeCounter + 1)
+        },
+        [savedViewChangeCounter]
+    );
+
+    const saveSearch = useCallback(
+        async (saveCurrentName?: boolean) => {
+            if (saveCurrentName) {
+                const viewJson = JSON.stringify({ filters: getSearchFilterObjs(filters) });
+                await saveFinderSearch(currentView, viewJson, saveCurrentName);
+                setViewDirty(false);
+            } else {
+                setShowSaveViewDialog(true);
+            }
+        },
+        [currentView, filters]
+    );
+
+    const searchViewJson = useMemo(() => {
```

Comment:
should this be moved up so that it can be used within `saveSearch`, since that looks to be doing the same JSON.stringify() call?